### DISCUSSION TITLE
etrade transactions

### DISCUSSION THREAD

**Discussion by 1rocketdude** [jessecooper/pyetrade#60] [Ideas]
is anyone interested in being able to grab transactions from the API?
I have always relied on the eTrade web site to download my trades as a CSV file, but that may not have all the information that people are looking for in their book-keeping or back-testing.

I can do it pretty easily, but wondering if others are interested.

**Comment by jessecooper** [comment]
https://github.com/jessecooper/pyetrade/blob/master/pyetrade/accounts.py#L143

I have all the transactions API done in the accounts object. Should be good to go.

**Comment by CaptainZ8** [comment]
Thank you Jesse, that is very helpful.

One question: I was looking for the E*Trade Steaming API, so I can listen
to order status updates and executions, but couldn’t find any. Do you have
any insight on this?


Best,


Zhen

**Reply by jessecooper** [reply]
In v0 they had a stream API I never got around to implementing against. I don't think they have it anymore in v1 as I no longer see it mentioned in the docs.

**Comment by kgstorm** [comment]
Hey all.
Just in case anyone else runs into this…
I was trying to use the kwarg startDate by itself to get transactions from today, but the request was always ignored. I find that to use startDate, you must also use endDate. Maybe I missed it, but didn’t see that in the pyetrade or eTrade API docs.
Cheers
Kevin

**Comment by azzugit** [comment]
Hi @1rocketdude , did you figure out out to get the transactions via API calls ?, I am trying to do the same thing but hitting a wall. I am using this code

--------------------
import pyetrade

consumer_key = "XXX"
consumer_secret = "XXXX"
tokens = {'oauth_token': 'xxxx', 'oauth_token_secret': 'xxxxx'}
accounts = pyetrade.ETradeAccounts(
    consumer_key,
    consumer_secret,
    tokens['oauth_token'],
    tokens['oauth_token_secret']
    
)

print(accounts.list_transactions("account_ID", resp_format='json'))

**Comment by 1rocketdude** [comment]
This works for me; steps are 1. Auth, 2. Get accounts, 3. List transactions for a particular account

I think you are just missing the middle step

# get the account credentials and log in
consumer_key = ‘Etrade_consumer_key'
consumer_secret = 'Etrade_consumer_secret'
oauth = pyetrade.ETradeOAuth(consumer_key, consumer_secret)
auth_url = oauth.get_request_token()

API_token = input('eTrade token: ')
oauth.get_access_token(API_token)
accounts_api = pyetrade.accounts.ETradeAccounts(consumer_key, consumer_secret, oauth.access_token['oauth_token'],
oauth.access_token['oauth_token_secret'], dev=False)
acc = accounts_api.list_accounts('json’)
my_account = ‘123456xxx'

trans = accounts_api.list_transactions(idkeys[my_account],'json')
transactions = trans['TransactionListResponse']
with open('transactions.json','wt') as fout:
json.dump(transactions, fout, indent=2)

**Comment by Chi-Square** [comment]
Has anyone tried to pull data into google sheets as a feed using this API?
